Violin.m

```diff
@@ -1,714 +1,726 @@
+                jitterstrength = density*width;
+            end
+            if isempty(data2) % if no comparison data
+                jitter = 2*(rand(size(data))-0.5); % both sides
+            else
+                jitter = rand(size(data)); % only right side
+            end
+            switch args.HalfViolin % this is more modular
+                case 'left'
+                    jitter = -1*(rand(size(data))); %left
+                case 'right'
+                    jitter = 1*(rand(size(data))); %right
+                case 'full'
+                    jitter = 2*(rand(size(data))-0.5);
+            end
+            % Make scatter plot
+            switch args.DataStyle
+                case 'scatter'
+                    if ~isempty(data2)
+                        jitter = 1*(rand(size(data))); %right
+                        [x,y] = obj.plotOrientation(pos + jitter.*jitterstrength, data);
+                        obj.ScatterPlot = ...
+                            scatter(x, y, args.MarkerSize, 'filled');
+                        % plot the data points within the violin area
+                        if length(densityC) > 1
+                            jitterstrength = interp1(valueC, densityC*widthC, data2);
+                        else % all data is identical:
+                            jitterstrength = densityC*widthC;
+                        end
+                        jitter = -1*rand(size(data2));% left
+                        [x,y] = obj.plotOrientation(pos + jitter.*jitterstrength,data2);
+                        obj.ScatterPlot2 = ...
+                            scatter(x, y, args.MarkerSize, 'filled');         
+                    else
+                        [x,y] = obj.plotOrientation(pos + jitter.*jitterstrength,data);
+                        obj.ScatterPlot = ...
+                            scatter(x, y, args.MarkerSize, 'filled');
+
+                    end
+                case 'histogram'
+                    [counts,edges] = histcounts(data, size(unique(data),1));
+                    switch args.HalfViolin
+                        case 'right'
+                            [x,y] = obj.plotOrientation([pos-((counts')/max(counts))*max(jitterstrength)*2, pos*ones(size(counts,2),1)]',...
+                                        [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2]);    
+                        case 'left'
+                            [x,y] = obj.plotOrientation([pos*ones(size(counts,2),1), pos+((counts')/max(counts))*max(jitterstrength)*2]', ...
+                                        [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2]);
+                        otherwise
+                            fprintf([namefun, ' No histogram/bar plot option available for full violins, as it would look overcrowded.\n'])
+                    end
+                    obj.HistogramPlot = plot(x,y,'-','LineWidth',1, 'Color', 'k');
```

Comment:
This is referring to an earlier comment.

Here, the `case`s no longer plot, but merely prepare an `x` and `y`. This makes it harder to understand what's going on. I'd prefer the plot commands to still be inlined, as this is explicitly what the code does: It plots things. Preparing data is merely an ancillary complexity for the dual-orientation use case.

This is exacerbated by the name `plotOrientation`, which implies that the function actually plots. Which it doesn't. It merely swaps two variables.

I'd prefer a structure like this in every `case` instead:

```
x = [pos*ones(size(counts,2),1), pos+((counts')/max(counts))*max(jitterstrength)*2]';
y = [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2];
[x, y] = obj.swapOrientationsMaybe(x, y);
obj.HistogramPlot = plot(x, y, '-', 'LineWidth', 1, 'Color', 'k');
```

Our first objective here is readability and maintainability!

Reply:
I would use the above function name `swapOrientationsMaybe`. But feel free to use a different name if you find one.